modules/ListItem.js

```diff
@@ -62,24 +63,40 @@ const styles = StyleSheet.create({
     },
 });
 
-const ListItem = (props) =>{
-    return(
-        <View style={styles.container}>
-            <Image style={styles.image}
-                source={{uri: props.imageUrl}}
-            />
-            <View style={styles.details}>
-                <View style={styles.upperDetails}>
-                    <Text style={styles.name}>{props.name}</Text>
-                    <View style={styles.idView}>
-                        <Text style={styles.id}>{props.id}</Text>
+class ListItem extends Component{
+    constructor(props){
+        super(props);
+        this.onPressHandler = this.onPressHandler.bind(this);
+    }
+
+    onPressHandler(){
+        try{
```

Comment:
Here you use `try catch` but on the parent component u use `then catch`. Please be consistent in the code writing.